src/main/java/org/sonarlint/intellij/util/SonarLintAppUtils.java

```diff
@@ -35,11 +36,26 @@
 import java.util.Arrays;
 import javax.annotation.CheckForNull;
 
+import static com.intellij.openapi.application.ApplicationManager.getApplication;
+
 public class SonarLintAppUtils extends ApplicationComponent.Adapter {
   @CheckForNull
   public Module findModuleForFile(VirtualFile file, Project project) {
-    ApplicationManager.getApplication().assertReadAccessAllowed();
-    return ProjectFileIndex.SERVICE.getInstance(project).getModuleForFile(file, false);
+    return runReadAction(() -> {
+      if (!project.isOpen()) {
+        return null;
+      }
+      return ProjectFileIndex.SERVICE.getInstance(project).getModuleForFile(file, false);
+    });
+  }
+
+  private static <G> G runReadAction(Computable<G> computable) {
+    Application application = getApplication();
+    if (application == null || application.isReadAccessAllowed()) {
```

Comment:
What is the purpose of `application.isReadAccessAllowed` here? IIRC it gets called by `application.runReadAction` anyway.

Reply:
I found this pattern in multiple places in the code, so I though it was a good idea. But your're right, this is useless.